active/ItemsSourceView-Indexof/ItemsSourceView-IndexOf.md

```diff
@@ -0,0 +1,67 @@
+# Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+
+Currently the ItemsSourceView class does not expose an IndexOf function. However during development, this method was used multiple times internally on derivatives of the class. To make the usage of this existing function easier on subclasses, we should expose this function in the ItemsSourceView.
```

Comment:
I can't remember, why isn't ItemsSourceView just an IVector?

Reply:
I think the direction IVector is going and where ItemsSourceView is used are slightly different. The idea of IVector is to enable different collection operations, while ItemsSourceView seems to aim to provide a way to make all kind of collections accessible to the ItemsRepeater and usable for rendering. Adding and removing items wouldn't be an operation that we need in that case I think. But that's just my guess here.

Reply:
Actually I should have said [IVectorView<T>](https://docs.microsoft.com/uwp/api/Windows.Foundation.Collections.IVectorView_T_). With IndexOf added, the only thing ItemsSourceView won't have is GetMany. Not a big problem, just would have been cleaner if we'd done it that way in the first place.

I don't think it's worth adding it now though, it's better to do it as you have it. The weird thing about doing it now is that it currently has a Count property, which matches [IReadOnlyCollection<T>.Count](https://docs.microsoft.com/dotnet/api/System.Collections.Generic.IReadOnlyCollection-1.Count), but not [IVectorView<T>.Size](https://docs.microsoft.com/uwp/api/Windows.Foundation.Collections.IVectorView-1.Size), so it would just be adding some confusion.

Reply:
Oh yes, that makes more sense. I guess there was no value in adding that previously (maybe the method was not needed internally?). The missmatch between Count and Size that we would create is probably not worth the benefit of having ItemsSourceView implement IVectorView.